### PR TITLE
UX improvements

### PR DESCRIPTION
**Pull request by ladyron**
# New features:

- Toolbar menu at the top of the document with shortcuts to common LaTeX editing tasks
  image
  - Preview Button (easier way to trigger LaTeX compilation; currently triggered by clicking 'Show LaTeX Preview' in the context menu)
  - Typeface Modifier GUI (subscript, superscript, bold, italic, underlined)
  - Text Layout Customization GUI: This feature allows the user to format a paragraphs layout (alignment for example)
  - Add plot (from list of canned `tikzplots` templates): simple function plot, plot from file, scatter plot, bar graphs, contour plot, parametric plot
  - Table Creation GUI (select number of rows and columns to create a table with dummy data)
  - Bullet Point and Numbered List

- Error Log Filtering options
  

- Main menu section with additional helpers
  - Constant Menu
  - Symbol Menu

- Tab-autocomplete for common LaTeX tags

Test in Binder:
[Binder](https://mybinder.org/v2/gh/Cbarr-hub/jupyterlab-latex/development)

# About the team
This PR is implemented by Team 1318, a Georgia Tech Junior Design team.

# User research
The added features were selected as a result of the user research amongst NIH community.

- People from the following fields were surveyed:
  - Software Engineering
  - Data Science
  - Cheminformatics
  - Virology
  - Drug discovery
  - Bioinformatics
  - Student

image

image

image

image

image

image

image

- What was the most difficult about using LaTeX?
  - The lack of the good editor
  - Memorizing settings that has not been used for long time
  - Layout
  - Formatting of things that aren't text
  - Sharing docs with colleagues
  - Lack of rich features present in MS word
  - Table generation
  - Graph creation

- Features that users would like to see within JupyterLab-LaTeX?
  - Table generator
  - Equation generation
  - Literate programming like environment (e.g., cweb)
  - Real time collaboration
  - GUI for tables and figures would be amazing
  - Easy/simple simultaneous render of formatted document and what is being typed
  - Robust templates for journals to "auto apply" pre-review AND post-review formatting required

### CODE REVIEW

**Comment by ktaletsk** [comment]
`Untitled.ipynb` and `.DS_Store` should be removed. I recommend updating .giignore to filter .DS_Store files

File: src/index.ts [hunk 1]
```diff
@@ -386,6 +393,249 @@ function activateLatexPlugin(
...
+
+      const underlineButton = new ToolbarButton({
+        className: 'underline-text',
+        label: 'U',
+        onClick: insertUnderline,
+        tooltip: 'Click to insert underlined text'
+      });
+      const bulletListButton = new ToolbarButton({
+        className: 'insert-bullet-list',
+        label: 'Bullet List',
```
[the remark this answers is not in the document]
Comment (Cbarr-hub) [reply src/index.ts]: Resolved

File: src/index.ts [hunk 2]
```diff
@@ -386,6 +393,249 @@ function activateLatexPlugin(
...
+        onClick: insertBulletList,
+        tooltip: 'Click to insert bullet list'
+      });
+
+      const numberedListButton = new ToolbarButton({
+        className: 'insert-numbered-list',
+        label: 'Numbered List',
```
[the remark this answers is not in the document]
Comment (Cbarr-hub) [reply src/index.ts]: Resolved

File: src/index.ts [hunk 3]
```diff
@@ -50,7 +50,15 @@ import { IMainMenu } from '@jupyterlab/mainmenu';
 
 import latexIconStr from '../style/latex.svg';
 
+import { InputDialog } from '@jupyterlab/apputils';
```
[the remark this answers is not in the document]
Comment (Cbarr-hub) [reply src/index.ts]: resolved

File: src/index.ts [hunk 4]
```diff
@@ -386,6 +393,249 @@ function activateLatexPlugin(
     state.save(id, { paths: Array.from(Private.previews) });
   };
 
+  class EditorButtonExtension
```
[the remark this answers is not in the document]
Comment (Cbarr-hub) [reply src/index.ts]: resolved

File: src/index.ts [hunk 5]
```diff
@@ -386,6 +393,249 @@ function activateLatexPlugin(
...
+              if (widget) {
+                let editor = widget.content.editor;
+                if (editor.replaceSelection) {
+                  editor.replaceSelection('_{' + value.value + '}');
+                }
+              }
+            }
+          }
+        );
+      };
```
[the remark this answers is not in the document]
Comment (Cbarr-hub) [reply src/index.ts]: resolved

File: src/index.ts [hunk 6]
```diff
@@ -642,6 +893,171 @@ function addSynctexCommands(
...
+  constants.set('Golden Ratio', '\\varphi');
+
+  constants.forEach((value: string, key: string) => {
+    let commandName = 'latex:' + key.replace(' ', '-').toLowerCase();
+    app.commands.addCommand(commandName, {
+      label: key,
+      caption: value,
+      execute: async args => {
+        let widget = editorTracker.currentWidget;
+        if (widget && PathExt.extname(widget.context.path) === '.tex') {
```
[the remark this answers is not in the document]
Comment (Cbarr-hub) [reply src/index.ts]: resolved

File: README.md [hunk 7]
```diff
@@ -114,4 +114,94 @@ c.LatexConfig.shell_escape = "allow"
...
+### Bug Fixes
+
+- LaTeX Editor toolbar buttons are now only visible for .TEX files
+
+### Known Issues
+
+N/A
+
+---
+
```
[the remark this answers is not in the document]
Comment (Cbarr-hub) [reply README.md]: resolved

**Pull request merged**